### DISCUSSION TITLE
I2C Pins

### DISCUSSION THREAD

**Discussion by randel0** [hatlabs/discussions#21] [General]
Is there a reason that SDA and SCL aren't on pins 21 and 22?

It took a little bit of troubleshooting to discover this, and this doesn't seem to be covered in the documentation.

**Comment by mairas** [comment]
When the board was being designed, there was a plan to create an Ethernet add-on hat for the SH-ESP32. The hat hasn't materialized yet, but the provision had some impact on the peripheral pinouts. If you look at the 14 rightmost pins on the GPIO header, they're clustered there just for that purpose.

The pinouts are listed in this section:

https://hatlabs.github.io/sh-esp32/pages/hardware/#pinouts-of-peripherals

Do you think it'd be helpful to additionally scatter the information in the individual peripheral descriptions as well?

**Reply by mairas** [reply]
Here's the link to the Ethernet hat design. I'm not sure what state the design is in - at least no proto orders were ever made, so I'm pretty sure there're still some dealbreaker bugs around.

https://github.com/markfarnan/SH-ESP32-Ethernet

**Reply by AAlbero** [reply]
Very annoying. Now I have to change all programs to accommodate this design flaw, caused by something that does not exist. Ouch. 
Oh no, I can't. Because now I have to have different programs for different boards. NICE.

**Reply by mairas** [reply]
The ESP32 platform doesn't have many hardcoded digital IO GPIOs, so you always need to verify the pinout on a new board.

You can store the pin definitions as constants in `platformio.ini` and then conditionally define the pins according to the board you're using. It's quite nice, really. :-)

**Reply by AAlbero** [reply]
Yes I am learning that. I am so impressed by all the work being done in this project. I wish I could play with this all day every day. Once I have OTA working reliably, I will be able to better maintain those ESP32's in a short leash.
Thank you so much for all you have done. Incredible. Now if I can just make PlatformIO stop breaking I will be very happy. Or back to the old Arduino IDE.

**Comment by AAlbero** [comment]
@mairas maybe you can help me id a curious issue I am having. 
I placed an I2c bus with 5 INA226. When I use the usual 21/22 pin combination, I can see and properly get data from all 5 devices.
When I use the built in I2C header (pins 16/17), I can only see 4 devices. if I disconnect one, I still see 4, being that the one that was "missing" now shows up. This happens regardless of what device I connect/disconnect.
I am using 4.7k pullup resistors in my setup, so maybe the SH-ESP32 I2C header already have pull up resistors there??

**Reply by AAlbero** [reply]
OK. I see in the schematics that the SH already has 2k pull up resistors there. Maybe that is why. I will try a variation in my setup with 2.7k resistors

**Reply by AAlbero** [reply]
FYI, removing my 4.7K resistors solved the issue with the built in I2C header.

**Reply by mairas** [reply]
That's interesting - adding an extra pull-up shouldn't really affect anything. It just makes the pull-up a bit stronger, but I wouldn't have assumed that would've caused any issues. But glad you got it sorted out.

And that's quite a few ADC channels as well! 😄
